### PR TITLE
feat: Arithmetics

### PR DESCRIPTION
**Pull request by rsanjuan87**
Feature to be abble calc some arithmetics functions right in the launcher

Log:

### CODE REVIEW

File: src/windowedframe.cpp [hunk 1]
```diff
@@ -894,7 +894,7 @@ void WindowedFrame::searchText(const QString &text)
             m_focusPos = Search;
         }
         auto temp = text;
-        m_appsManager->searchApp(temp.replace(" ",""));
+        m_appsManager->searchApp(temp.trimmed());
```
Comment (wentaosong1993) [inline src/windowedframe.cpp:897]: trimmed : Returns a string that has whitespace removed from the start and the end. 
replace: Replaces every occurrence of the string before with the string after and returns a reference to this string.

**Review by wentaosong1993 — COMMENTED** [review]
As far as I'm concerned, as for this commit, I think it's good work, but this extended function does't fit with current functional demand.

File: src/windowedframe.cpp (hunk 1 above)
Comment (wentaosong1993) [reply src/windowedframe.cpp:897]: > for real, I don't remember why I did that, almost a year old or more

I'm sorry for that, maybe we need time to adjust our work to keep up with your pace. We will review the code as much as we can.

**Comment by rsanjuan87** [comment]
> As far as I'm concerned, as for this commit, I think it's good work, but this extended function does't fit with current functional demand.

in facts this PR adds a few features more than aritmetics

- Locating Folder 
- Option to show/select in the dde-file-manager
-  show/run option in terminal
-  element actions (open with, show in folder or shell)
-  option to search the web

File: src/windowedframe.cpp (hunk 1 above)
Comment (rsanjuan87) [reply src/windowedframe.cpp:897]: > trimmed : Returns a string that has whitespace removed from the start and the end. replace: Replaces every occurrence of the string before with the string after and returns a reference to this string.

I think it comes from here
https://github.com/linuxdeepin/dde-launcher/pull/95/commits/463ca29e519b50f0c41ac317e02edf8b98be8eda

**Comment by BLumia** [comment]
I'd like to close this one for now since `dde-launcher` will be deprecated in next upcoming release. It will be replaced by `dde-launchpad`.

But for your feature request, we suggest you use [`dde-grand-search`](https://github.com/linuxdeepin/dde-grand-search/) for this purpose. It is similar to macOS' spotlight and more suitable for this feature, and it is extendable by writing plugin :)

**Pull request closed without merging**